ucfcs/GrowthPlus
# Julio/child screen progress

This PR implements child progress in parent portal and child screen. 

To test: 
- Uncomment line 70 in MainActivity, it will import the sample child Lili.
- Do a fresh rebuild of the app. 
- Create a parent, the child Lili will be imported from realm. 
- Navigate to parent portal, Lili's vertical progress bar will reflect the number of lesson completed across the app, `(numCompleted/ totalLessons) * 100` . 
- Tap on the child card and navigate to child screen, each horizontal bar represents the number of lessons completed in each roadmap, ensure the lessons categories are populated correctly, some will have a green checkmark and most will be red x's. 
- Navigate back to language settings and change the language, navigate back to child screen and inspect that the category names translate correctly. 

Note: 
If you create another child, the progress reflected in parent portal and child screen will not be accurate, these variables are automatically set to 0 and are implemented in such way that only a natural progression will reflect accurate scores, it can't be hardcoded except with realm importer. We'll soon set up the state of the child to start at the very beginning as it should be.
Some clean up and variable re-structure was implemented as well, this doesn't change any logic, it is mostly for readability and to get rid of android studio warnings.
